### DISCUSSION TITLE
Eduponics with ESPHome

### DISCUSSION THREAD

**Discussion by csthomas2** [STEMinds/STEMinds-Docs#11] [Show and tell]
Hi everyone. 

I just wanted to share my recent adventures with my Eduponics Mini. I thought it would be an interesting experience to see if I could get the Eduponics board to integrate with Home Assistant since I already had a number of integrations setup including smart switches to automatically control the lights for the plants. The Eduponics Mini board was the perfect addition.

ESPHome already had built in libraries for the popular sensors on the main board. After reading through the documentation and Python/Arduino examples, the main board was up and running smoothly. This personal project included 4 of the soil sensors though, so getting the expansion board setup took a little while longer.

ESPHome does include libraries for the MCP23017 and the ADS1115. This made things a bit easier. Turning the expansion board on reliably took a little tweaking. The major hurdle proved to be troubleshooting power. It seems that the ESPHome firmware does draw more power than expected. Whenever I would turn on a soil moisture sensor, the board would reset and the ESP32 would lock up. I finally put a 10uf cap between 5V and ground on the output of the expansion board. This seems to provide enough of a boost to turn the sensor on and has been rock solid.

I'm currently using Node-Red to send notifications to my phones and dashboards through the Home Assistant API as I collect metrics to find which values are ideal thresholds for when to begin watering and how long to run the pump. Node-Red will handle that automation.
![image](https://user-images.githubusercontent.com/62066498/213828524-601586b4-4915-4f6f-90a5-eea17378e41e.png)

The ESPHome configuration was written to include safeties for the pumps though. It will automatically turn the relays off if the water level runs too low and will not allow it to be turned back on.

The configuration creates entities in Home Assistant for all the sensors (light, temperature, humidity, water level, and air pressure) on the main board. It also creates entitles for the soil moisture sensor (calibration is in the config) and pump. Since I have the extension board, I also created entities for the 4 moisture sensors and 4 relay outputs. The ESPHome configuration handles the automation of toggling the MCP Pins then taking the reading and converting to a percentage through calibration.

This board is amazing and a perfect addition to any garden, so I created a gist to share and hopefully help anyone else who is looking to use it with Home Assistant get started.

https://gist.github.com/csthomas2/38a6959f7ba55796f8adc45474220743

edit: updated gist link, spelling

**Comment by Zontex** [comment]
Brilliant, @csthomas2  would you mind if I share about your project in the next Eduponics v2.0 campaign update?
The 10uF cap sounds interesting. I'll add the fix in the new hardware!

**Comment by csthomas2** [comment]
Thanks. Sure, no problem.
